Engine Hacks/Necessary/CalcLoops/CanUnitDoubleCalcLoop/CanUnitDoubleCalcLoop.s

```diff
@@ -0,0 +1,276 @@
+.thumb
+.align
+
+.global CanUnitDoubleCalcLoopFunc
+.type CanUnitDoubleCalcLoopFunc, %function
+
+
+.equ Get_Weapon_Effect, 0x8017724
+
+
+CanUnitDoubleCalcLoopFunc:
+
+@we need a good way to do this
+@functions in the loop can either 
+@A. set to always true
+@B. set to always false
+@C. keep the same as before
+
+@they need passed attacker & defender
+@the teq func for this liked to load the attacker & defender structs manually
+@but they get passed in via the stack
+@(pointers to via stack or them loaded into the stack?)
+
+@seems to be pointers to via stack
```

Comment:
On the attack preview drawing function, the correct values aren't yet in these stack pointers (instead it's some leftover stuff from text draw functions). This causes incorrect doubling data to show in the battle preview. Recommend changing these to:

```
ldr r4,=#0x203A4EC @attacker struct
ldr r5,=#0x203A56C @defender struct
```